### ISSUE TITLE
BLTouch support

### ISSUE THREAD

**Issue by Frix-x** [Frix-x/klippain#296]
### Describe the feature or hardware support you'd like

This is asked quite frequently on Discord and would help a lot of people setting up Klippain on their machines

**Comment by SudoFi** [comment]
I took a stab at this and wasn't having much luck. Essentially one could just put all the bltouch config into the override and it would accept it, since its natively supported by klipper:

```
[bltouch]
sensor_pin:
control_pin:  
x_offset: 
y_offset: 
z_offset: 
samples: 
samples_result: 
sample_retract_dist: 
samples_tolerance: 
samples_tolerance_retries: 

[stepper_z]
endstop_pin: probe:z_virtual_endstop
```
I attempted to do this by adding a probe config _(/config/hardware/probes/bltouch.cfg)_. 
Then adding a inclusion to printer.cfg pointing to the probe config. 

Including very basic configuration in _generic_probe.cfg_:
`_CHECK_PROBE action=query`

Plugging variables "PROBE_INPUT" _(for sensor_pin)_ and "SERVO_PIN" _(for control_pin)_ into the config. I would get error : _Pin 'SERVO_PIN' is not a valid pin name on mcu 'mcu''_.
If I override the control_pin and directly point it to the pin, it will work. 

Once I attempt to home Z, the probe will active but it retracts quickly after. Presenting with the error:
`Failed to verify BLTouch probe is raised; retrying.`
`BLTouch failed to raise probe`
Tried switching pullup or any other prefix to the pin yet is always states its "Open" never triggered. 
Including if I send commands to pin_up or pin_down. 
Im sure this is due t o my basic configuration in generic_probe.cfg. 

If I override both sensor_pin and control_pin, it works without a problem. 

I have a fork of all my changes. Please let me know your recommendations. 
https://github.com/SudoFi/klippain-bltouch

**Comment by SudoFi** [comment]
> ### Describe the feature or hardware support you'd like

under my fork, in variables.cfg, I added the majority BLTouch Variables Klipper supports. 

```
variable_probe_bltouch_pin_up_touch_mode_reports_triggered
variable_probe_bltouch_pin_up_touch_mode_reports_not_triggered
variable_probe_bltouch_probe_with_touch_mode
variable_probe_bltouch_stow_on_each_sample
varibale_probe_bltouch_store_mode_5v
```
I would say these would probably be the most likely features that would need support.

**Comment by Surion79** [comment]
@SudoFi I looked in your fork and my main question would be: where are the variables used? i see the variables but no functionality where they could be used. The rest looks good and i could do a PR. But the variable usage is unclear for me, since it is not shown in your fork.

**Comment by nmsmith22389** [comment]
Any progress on this?

**Comment by tristanmcpherson** [comment]
> Any progress on this?

I have this working. I'll clean up my example and submit a PR

**Comment by tristanmcpherson** [comment]
> Was just looking at this and preparing a PR at the same time as Tristan, will wait for his. The version by @SudoFi seemed almost OK - had an additional colon in bltouch.cfg line 15, which is why it didn't work.
> 
> Also, as per the question of @Surion79, perhaps the variables should be removed entirely, if one needs to change the behavior then overrides can be used, no need to over complicate things.

I agree here. The variables are nice in case of macros for things that may want to be configured per material, etc. In more of a dynamic way. I think using the overrides is a more sane way here

**Comment by SudoFi** [comment]
I apologize for the delayed response, work has keep me too busy for some of my hobbies.

@tristanmcpherson , @Surion79 @agios @nmsmith22389 
So I ended up going the same route you had mentioned @tristanmcpherson. I just added all my changes into "overrides" for the same reasons. There is no need to complicate things if there is a solution that already works. Since many still use BLtouch, it may be worth adding a commented out section to overrides with the essentials for users. @Frix-x
